newIDE/app/src/MainFrame/EditorTabsHandler.js

```diff
@@ -130,6 +130,22 @@ export const closeProjectTabs = (
   );
 };
 
+/*
+ * Ask the editors to persist their UI settings
+ * to the project.
+ */
+export const saveUiSettings = (state: EditorTabsState) => {
+  state.editors.forEach(editorTab => {
+    if (
+      editorTab.editorRef &&
+      editorTab.editorRef.editor &&
```

Comment:
Ok I understood, it's because here you're asking the editor to have itself a ref called `editor` that has itself a method called saveUiSettings. We're going one way to deep, breaking the encapsulation of `SceneEditorContainer` (which is in `newIDE/app/src/MainFrame/Editors/SceneEditor.js`). So we should:

* Change this to  `if (editorTab.editorRef && editorTab.editorRef.saveUiSettings) { editorTab.editorRef.saveUiSettings())`
  * (see how conveniently we don't have to dig into components by doign a.b.c.d(). Doing more than one level of `a.b.c` is usually the sign that some methods are missing).
* Move saveUiSettings in SceneEditorContainer
* Also put it in ExternalLayoutEditor (in newIDE/app/src/MainFrame/Editors/ExternalLayoutEditor.js)

Note that both of these editors don't have a this.props.layout. You have to use `this.getLayout`. Knowing that it can return null, so you should `const layout = this.getLayout()` and check if it's not null before calling unserializeFromJSObject.

Reply:
Ok, I will look into refactoring it again once I get home :)

Reply:
Thanks! Once we get that working it should be good to go.

Reply:
ok it's ready now :) 🚀

Reply:
ugh, need to fix flow errors again

Reply:
fixed 👍